qiskit/passmanager/passmanager.py

```diff
@@ -12,183 +12,203 @@
+            tasks = _legacy_build_flow_controller(
+                tasks,
+                options={"max_iteration": self.max_iteration},
+                **flow_controller_conditions,
+            )
+        if isinstance(tasks, Sequence):
+            tasks = FlowControllerLinear(tasks)
+        self._flow_controller.tasks += (tasks,)
 
     def replace(
         self,
         index: int,
-        passes: PassSequence,
+        tasks: Task | list[Task],
         **flow_controller_conditions: Any,
     ) -> None:
         """Replace a particular pass in the scheduler.
 
         Args:
             index: Pass index to replace, based on the position in passes().
-            passes: A pass set (as defined in :py:func:`qiskit.transpiler.PassManager.append`)
-                to be added to the pass manager schedule.
-            flow_controller_conditions: control flow plugins.
+            tasks: A set of pass manager tasks to be added to schedule. When multiple
+                tasks are provided, tasks are grouped together as a single flow controller.
+            flow_controller_conditions: Dictionary of control flow plugins.
+                See :meth:`~.BasePassManager.append` for details.
 
         Raises:
-            PassManagerError: if a pass in passes is not a proper pass or index not found.
+            PassManagerError: If the index is not found.
         """
-        passes = self._normalize_passes(passes)
-
+        if flow_controller_conditions:
+            tasks = _legacy_build_flow_controller(
+                tasks,
+                options={"max_iteration": self.max_iteration},
+                **flow_controller_conditions,
+            )
+        if isinstance(tasks, Sequence):
+            tasks = FlowControllerLinear(tasks)
         try:
-            self._pass_sets[index] = {
-                "passes": passes,
-                "flow_controllers": flow_controller_conditions,
-            }
+            new_tasks = list(self._flow_controller.tasks)
+            new_tasks[index] = tasks
+            self._flow_controller.tasks = tuple(new_tasks)
         except IndexError as ex:
             raise PassManagerError(f"Index to replace {index} does not exists") from ex
 
     def remove(self, index: int) -> None:
         """Removes a particular pass in the scheduler.
 
         Args:
-            index: Pass index to replace, based on the position in passes().
+            index: Pass index to remove, based on the position in passes().
 
         Raises:
-            PassManagerError: if the index is not found.
+            PassManagerError: If the index is not found.
         """
         try:
-            del self._pass_sets[index]
+            new_tasks = list(self._flow_controller.tasks)
+            del new_tasks[index]
+            self._flow_controller.tasks = new_tasks
         except IndexError as ex:
             raise PassManagerError(f"Index to replace {index} does not exists") from ex
```

Comment:
This is another place where trying to maintain the inner `FlowControllerLinear` (which isn't supposed to be mutable) is causing problems, and it'd be cleaner just to create that controller during `run`.